app/controllers/api/authentications_controller.rb

```diff
@@ -1,4 +1,27 @@
 module Api
   class AuthenticationsController < BaseController
+    def create_resource(_type, _id, data)
+      attrs = validate_auth_attrs(data)
+      klass = ::Authentication.class_from_request_data(attrs)
+      # TODO: Temporary validation - remove
+      raise 'type not currently supported' unless klass.respond_to?(:create_in_provider_queue)
+      task_id = klass.create_in_provider_queue(attrs['manager_resource'], attrs.except('type', 'manager_resource'))
+      action_result(true, 'Creating Authentication', :task_id => task_id)
+    rescue => err
+      action_result(false, err.to_s)
+    end
+
+    private
+
+    def validate_auth_attrs(data)
+      raise 'must supply a manager resource' unless data['manager_resource']
+      attrs = data.dup
+      attrs['manager_resource'] = if data['manager_resource'].key?('href')
+                                    parse_href(data['manager_resource']['href']).last
+                                  elsif data['manager_resource'].key?('id')
+                                    data['manager_resource']['id']
+                                  end
```

Comment:
A credential is an `Authentication` and is related to a `manager` through the polymorphic `resource_type/resource_id`.  So in some other user cases, an `Authentication` is related to other things, e.g. a `Vm`. 

In our use case here in `create_in_provider[_queue]`, it has to be a `manager` aka `ExtManagementSystem`. 

Yes, we can have `create_in_provider[_queue]` to take in a `manager`.  But I don't know if there's any other use case prefer the `id`.  @Fryguy or @bdunne ?